cylc/flow/post_install/log_vc_info.py

```diff
@@ -0,0 +1,285 @@
+        self.vcs = vcs
+        self.path = repo_path
+
+    def __str__(self) -> str:
+        return f"{self.vcs} repository at {self.path} is missing a base commit"
+
+
+def get_vc_info(path: Union[Path, str]) -> Optional['OrderedDict[str, str]']:
+    """Return the version control information for a repository, given its path.
+    """
+    info = OrderedDict()
+    missing_base = False
+    for vcs, args in VCInfo.INFO_COMMANDS.items():
+        try:
+            out = _run_cmd(vcs, args, cwd=path)
+        except VCSNotInstalledError as exc:
+            LOG.debug(exc)
+            continue
+        except VCSMissingBaseError as exc:
+            missing_base = True
+            LOG.debug(exc)
+        except OSError as exc:
+            if any(exc.strerror.lower().startswith(err)
+                   for err in VCInfo.NOT_REPO_ERRS[vcs]):
+                LOG.debug(f"Source dir {path} is not a {vcs} repository")
+                continue
+            else:
+                raise exc
+
+        info['version control system'] = vcs
+        if vcs == VCInfo.SVN:
+            info.update(_parse_svn_info(out))
+        elif vcs == VCInfo.GIT:
+            if not missing_base:
+                info['repository version'] = out.splitlines()[0]
+                info['commit'] = _get_git_commit(path)
+            info['working copy root path'] = str(path)
+        info['status'] = get_status(vcs, path)
+
+        LOG.debug(f"{vcs} repository detected")
+        return info
+
+    return None
+
+
+def _run_cmd(vcs: str, args: Iterable[str], cwd: Union[Path, str]) -> str:
+    """Run a command, return stdout.
+
+    Args:
+        vcs: The version control system.
+        args: The args to pass to the version control command.
+        cwd: Directory to run the command in.
+
+    Raises:
+        OSError: with stderr if non-zero return code.
+    """
+    cmd = [vcs, *args]
+    try:
+        proc = Popen(
+            cmd, cwd=cwd, stdin=DEVNULL, stdout=PIPE, stderr=PIPE, text=True)
+    except FileNotFoundError as exc:
+        # This will only be raised if the VCS command is not installed,
+        # otherwise Popen() will succeed with a non-zero return code
+        raise VCSNotInstalledError(vcs, exc)
+    ret_code = proc.wait()
+    out, err = proc.communicate()
+    if ret_code:
+        if any(err.lower().startswith(msg)
+               for msg in VCInfo.NO_BASE_ERRS[vcs]):
+            # No base commit in repo
+            raise VCSMissingBaseError(vcs, cwd)
+        raise OSError(ret_code, err)
+    return out
+
+
+def write_vc_info(
+    info: 'OrderedDict[str, str]', run_dir: Union[Path, str]
+) -> None:
+    """Write version control info to the workflow's vcs log dir.
+
+    Args:
+        info: The vcs info.
+        run_dir: The workflow run directory.
+    """
+    if not info:
+        raise ValueError("Nothing to write")
+    info_file = Path(run_dir, 'log', 'version', 'vcs.conf')
```

Comment:
I can't see a way to get the timestamp of the installation, perhaps this needs to be added to the entry point arguments?